src/utils/metrics/normalized_tags.rs

```diff
@@ -0,0 +1,163 @@
+use crate::{config::Config, utils::releases};
+use itertools::Itertools;
+use regex::Regex;
+use std::{borrow::Cow, cmp::min, collections::HashMap};
+
+pub(super) struct NormalizedTags<'a> {
+    tags: HashMap<Cow<'a, str>, String>,
+}
+
+impl<'a> From<&'a Vec<(String, String)>> for NormalizedTags<'a> {
```

Comment:
If I recall correctly, @Swatinem once told me that typically we use `From` to convert from an owned type. So, instead this should be 

```suggestion
impl<'a> From<Vec<(String, String)>> for NormalizedTags<'a> {
```

and the rest of the code should be changed correspondingly.

In any case, we should never take a reference to a `Vec`; instead, we should use a slice in such cases